optuna/visualization/_contour.py

```diff
@@ -226,16 +241,23 @@ def _get_contour_subplot(
         reversescale=reverse_scale,
     )
 
-    scatter = go.Scatter(
-        x=x_values,
-        y=y_values,
-        marker={"line": {"width": 2.0, "color": "Grey"}, "color": "black"},
+    return (
+        contour,
+        _create_scatter(feasible.x, feasible.y, "black"),
+        _create_scatter(infeasible.x, infeasible.y, "#cccccc"),
+    )
+
+
+def _create_scatter(x: list[Any], y: list[Any], marker_color: str) -> Scatter:
+    return go.Scatter(
+        x=x,
+        y=y,
+        marker={"line": {"width": 2.0, "color": "Grey"}, "color": marker_color},
```

Comment:
Here, `{"width": 2.0, "color": "Grey"}`, the line color of infeasible is also marker_color.